### ISSUE TITLE
Question about data preprocess

### ISSUE THREAD

**Issue by weizhepei** [truthless11/HRL-RE#2]
Hi, I notice that you filtered the dataset by "removing 1) the relations in ......; 2)the sentence that ...." as stated in the paper. And it says "Such a preprocess is also in line with the setting in the literature (for instance, **Tagging**)" . However, I read through the original paper (Zheng et al. 2017) and I can't find any similar description about the preprocessing. Did you confirm it with Zheng, or am I missing something? Could you please explain a little further? Thank you very much !!!

**Comment by truthless11** [comment]
It's trivial to filter out the relations that do not appear in the test set. You can see that all the sentences in the [test set](https://github.com/zsctju/triplets-extraction/tree/master/Data) from **Tagging** contain at least one relational fact, though they did not mention it explicitly in their paper. Also, **CopyR** divided the sentences into three types: Normal, EPO and SEO, and all the three types assume that each sentence has a relational triplet.

**Comment by weizhepei** [comment]
> It's trivial to filter out the relations that do not appear in the test set. You can see that all the sentences in the [test set](https://github.com/zsctju/triplets-extraction/tree/master/Data) from **Tagging** contain at least one relational fact, though they did not mention it explicitly in their paper. Also, **CopyR** divided the sentences into three types: Normal, EPO and SEO, and all the three types assume that each sentence has a relational triplet.

Thanks for your quick reply ! It's true that some preprocessing may not be explicitly mentioned in the paper, but I was just confused by the different results (on RE, NYT11) between zheng's and yours. If such a preprocessing is the same as zheng's , how could they got F1=0.495 (as reported in his paper) while you got 0.479 (see Table 2 of your paper) by using the same Tagging method? This might be stupid, but I want to make sure whether it is the different training dataset that leads to the different F1. That's why I asked "Did you confirm it with Zheng, or am I missing something? ". BTW, I could't achieve the same performance as he reported by directly running his released code. As a beginner, these are so weird to me...

**Comment by truthless11** [comment]
> > It's trivial to filter out the relations that do not appear in the test set. You can see that all the sentences in the [test set](https://github.com/zsctju/triplets-extraction/tree/master/Data) from **Tagging** contain at least one relational fact, though they did not mention it explicitly in their paper. Also, **CopyR** divided the sentences into three types: Normal, EPO and SEO, and all the three types assume that each sentence has a relational triplet.
> 
> Thanks for your quick reply ! It's true that some preprocessing may not be explicitly mentioned in the paper, but I was just confused by the different results (on RE, NYT11) between zheng's and yours. If such a preprocessing is the same as zheng's , how could they got F1=0.495 (as reported in his paper) while you got 0.479 (see Table 2 of your paper) by using the same Tagging method? This might be stupid, but I want to make sure whether it is the different training dataset that leads to the different F1. That's why I asked "Did you confirm it with Zheng, or am I missing something? ". BTW, I could't achieve the same performance as he reported by directly running his released code. As a beginner, these are so weird to me...

It's hard to say whether the different training sets or other tricks done by authors lead to the different F1 score, probably both do. Actually, we have run the released code on the original corpus too but did not get the same performance either. However, the result is quite comparable with what he reported in the paper. This is a common thing unless you get a far worse performance than reported.
